Config/FilterSetCollection.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the `liip/LiipImagineBundle` project.
+ *
+ * (c) https://github.com/liip/LiipImagineBundle/graphs/contributors
+ *
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Liip\ImagineBundle\Config;
+
+use Liip\ImagineBundle\Imagine\Filter\FilterConfiguration;
+
+final class FilterSetCollection
+{
+    /**
+     * @var array
+     */
+    private $filterSets = [];
+
+    /**
+     * @var FilterConfiguration
+     */
+    private $filterConfiguration;
+
+    /**
+     * @var FilterSetBuilderInterface
+     */
+    private $filterSetBuilder;
+
+    /**
+     * @param FilterConfiguration       $filterConfiguration
+     * @param FilterSetBuilderInterface $filterSetBuilder
```

Comment:
having factories for so many things, why do we not have a factory for the StackCollection? we could then define the public service for this as a thing that uses the StackCollectionFactory, and not have the value object that is used outside have the factory in it. the FilterSetBuilder could take that role, to not explode the number of infrastructure classes that we use.

for the symfony config, it would look like this: https://symfony.com/doc/current/service_container/factories.html